### PR TITLE
GitLab - Support imported pipelines

### PR DESCRIPTION
**Pull request by orizerah**
## Description
GitLab now supports imported pipelines (using the `include:` keyword)

## Checklist
- [x] I've read the [guidelines for contributing](https://github.com/argonsecurity/pipeline-parser/blob/main/CONTRIBUTING.md) to this repository.
- [x] I've followed the [conventions](https://github.com/argonsecurity/pipeline-parser/blob/main/CONTRIBUTING.md#pull-requests) in the PR title.
- [x] I've added tests that prove my fix is effective or that my feature works.
- [x] I've updated the [readme](https://github.com/argonsecurity/pipeline-parser/blob/main/README.md) with the relevant information (if needed).
- [x] I've added usage information (if the PR introduces new options)
- [x] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: pkg/parsers/azure/azure.go [hunk 1]
```diff
@@ -20,7 +20,7 @@ func (g *AzureParser) Parse(azurePipeline *azureModels.Pipeline) (*models.Pipeli
 	pipeline.Defaults = parsePipelineDefaults(azurePipeline)
 	pipeline.Triggers = parsePipelineTriggers(azurePipeline)
 	pipeline.Parameters = parseParameters(azurePipeline.Parameters)
-	pipeline.Imports = parseExtends(azurePipeline.Extends)
+	// pipeline.Imports = parseExtends(azurePipeline.Extends)
```
Comment (tamirkiviti13) [inline pkg/parsers/azure/azure.go:23]: Please remove.
Handle that in a future task

File: pkg/parsers/gitlab/imports.go [hunk 2]
```diff
@@ -1,27 +1,130 @@
...
-			imports = append(imports, item.Template)
-		}
-		if item.Local != "" {
-			imports = append(imports, item.Local)
-		}
-		if item.Remote != "" {
-			imports = append(imports, item.Remote)
-		}
-
+		imports = append(imports, parseIncludeItem(item))
```
Comment (tamirkiviti13) [inline pkg/parsers/gitlab/imports.go:25]: This function might return `nil`. 
Maybe it is better to check it first:
```suggestion
                 if importItem := parseIncludeItem(item); importItem != nil {
		        imports = append(imports, importItem)
		}
```

File: pkg/parsers/gitlab/imports.go [hunk 3]
```diff
@@ -1,27 +1,130 @@
...
+			Path:         &item.File,
+			Repository:   utils.GetPtr(splitProject[1]),
+			Organization: utils.GetPtr(splitProject[0]),
+		},
+		FileReference: item.FileReference,
+	}
+
+	if item.Ref != "" {
+		importData.Version = &item.Ref
+		importData.VersionType = models.BranchVersion
```
Comment (tamirkiviti13) [inline pkg/parsers/gitlab/imports.go:116]: Is it always of type `BranchVersion`?

File: pkg/parsers/gitlab/imports.go [hunk 4]
```diff
@@ -1,27 +1,130 @@
...
+
+	return importData
+}
+
+func parseTemplateImport(item *gitlabModels.IncludeItem) *models.Import {
+	if item.Template == "" {
+		return nil
+	}
+
+	fullTemplateUrl := fmt.Sprintf("https://gitlab.com/gitlab-org/gitlab/-/raw/master/lib/gitlab/ci/templates/%s", item.Template)
```
Comment (tamirkiviti13) [inline pkg/parsers/gitlab/imports.go:127]: Consider exporting it to a const

File: test/blackbox/utils.go [hunk 5]
```diff
@@ -1,21 +1,22 @@
...
 	githubEnhancer "github.com/argonsecurity/pipeline-parser/pkg/enhancers/github"
+	gitlabEnhancer "github.com/argonsecurity/pipeline-parser/pkg/enhancers/gitlab"
 	"github.com/argonsecurity/pipeline-parser/pkg/handler"
 	"github.com/argonsecurity/pipeline-parser/pkg/models"
 	"github.com/go-test/deep"
 )
 
 func readFile(filename string) []byte {
-	b, _ := ioutil.ReadFile(filename)
+	b, _ := os.ReadFile(filename)
```
Comment (tamirkiviti13) [inline test/blackbox/utils.go:19]: 😄

File: test/blackbox/utils.go [hunk 6]
```diff
@@ -25,6 +26,7 @@ func executeTestCases(t *testing.T, testCases []TestCase, folder string, platfor
 			h := http.FileServer(http.Dir(testCase.TestdataDir))
 			ts := httptest.NewServer(h)
 			githubEnhancer.GithubBaseURL = ts.URL
+			gitlabEnhancer.GITLAB_BASE_URL = ts.URL
```
Comment (tamirkiviti13) [inline test/blackbox/utils.go:29]: Sorry but it is really bothering me 😅
Please make it consistent (IDC which convention just stick to one)

File: pkg/enhancers/gitlab/gitlab.go [hunk 7]
```diff
@@ -1,16 +1,105 @@
...
+	if importData.Source.Type == models.SourceTypeLocal {
+		return handleLocalImport(importData)
+	}
+
 	return nil, nil
 }
 
+func handleRemoteImport(importData *models.Import, credentials *models.Credentials) (*enhancers.ImportedPipeline, error) {
+	if importData.Source.Type != models.SourceTypeRemote {
+		return nil, errors.New("invalid source type for remote import")
```
Comment (tamirkiviti13) [inline pkg/enhancers/gitlab/gitlab.go:54]: Nice :)

**Comment by tamirkiviti13** [comment]
Muy buen trabajo

File: pkg/parsers/gitlab/imports.go (hunk 3 above)
Comment (orizerah) [reply pkg/parsers/gitlab/imports.go]: Nope and fixed :)

**Review by tamirkiviti13 — APPROVED** [review]

**Pull request merged**